### DISCUSSION TITLE
useQuasar() inside a .ts file returning null

### DISCUSSION THREAD

**Discussion by brogits** [quasarframework/quasar#16995] [General - Components / Directives / etc] [answered]
Hi,

I had been noticing that there are times when useQuasar() is returning null when executed inside a .ts file

![image](https://github.com/quasarframework/quasar/assets/4569499/d9b6c773-794b-4887-959d-3a6af35ab3d4)

Line 324 still returns null even after executing useQuasar().

What could I be missing?

Is there standard way of doing this inside a .ts file?

Thanks

**Comment by rstoenescu** [comment, marked as answer]
If you're using `script setup` then you're ensured this error won't happen. However, otherwise just make sure that you are inside of your `setup()` function when calling it -- and calling it while setup() is running at runtime, not afterwards. This is just how composables work.

**Reply by brogits** [reply]
it was my bad, i actually need to export my service  with default class. thanks
